src/providers/wfs/qgsbackgroundcachedshareddata.cpp

```diff
@@ -227,8 +227,8 @@ bool QgsBackgroundCachedSharedData::createCache()
     return false;
   }
   const QString vsimemFilename = QStringLiteral( "/vsimem/qgis_cache_template_%1/features.sqlite" ).arg( reinterpret_cast< quintptr >( this ), QT_POINTER_SIZE * 2, 16, QLatin1Char( '0' ) );
-  mCacheTablename = CPLGetBasename( vsimemFilename.toStdString().c_str() );
-  VSIUnlink( vsimemFilename.toStdString().c_str() );
+  mCacheTablename = CPLGetBasename( vsimemFilename.toLocal8Bit().constData() );
```

Comment:
The use of toLcal8Bit() here is a bit mind-confusing, but I don't think it hurts as the filename must be ASCII only, so any encoding is fine. However I'd suggest the following for mental serenity:
```suggestion
  mCacheTablename = CPLGetBasename( vsimemFilename.toUtf8().constData() );
```